tests/test_base_plugin.py

```diff
@@ -0,0 +1,282 @@
+"""
+Copyright 2018, Oath Inc.
+Licensed under the terms of the Apache 2.0 license. See LICENSE file in project root for terms.
+"""
+import os
+import time
+import unittest
+
+from mock import patch, MagicMock
+from yapsy.PluginInfo import PluginInfo
+
+from yahoo_panoptes.framework.plugins.panoptes_base_plugin import PanoptesPluginInfo, PanoptesPluginInfoValidators, \
+    PanoptesPluginConfigurationError, PanoptesBasePluginValidators, PanoptesBasePlugin
+from yahoo_panoptes.polling.polling_plugin import PanoptesPollingPluginInfo
+from yahoo_panoptes.framework.resources import PanoptesResource, PanoptesContext
+from yahoo_panoptes.framework.utilities.helpers import get_module_mtime
+
+from .test_framework import PanoptesTestKeyValueStore, panoptes_mock_kazoo_client, panoptes_mock_redis_strict_client
+
+_TIMESTAMP = round(time.time(), 5)
+_LAST_EXECUTED_TEST_VALUE = 1458947997
+_LAST_RESULTS_TEST_VALUE = 1458948005
+
+mock_time = MagicMock(return_value=_TIMESTAMP)
+
+
+class TestPanoptesBasePluginSubclass(PanoptesBasePlugin):
+    def run(self, context):
+        pass
+
+
+class TestPanoptesBasePluginValidators(unittest.TestCase):
+    def test_valid_plugin_class(self):
+        self.assertFalse(PanoptesBasePluginValidators.valid_plugin_class(None))
+        self.assertFalse(PanoptesBasePluginValidators.valid_plugin_class(PanoptesContext))
+        self.assertTrue(PanoptesBasePluginValidators.valid_plugin_class(PanoptesBasePlugin))
+        self.assertTrue(PanoptesBasePluginValidators.valid_plugin_class(TestPanoptesBasePluginSubclass))
+
+
+class TestPanoptesPluginInfoValidators(unittest.TestCase):
+    def test_valid_plugin_info_class(self):
+        self.assertFalse(PanoptesPluginInfoValidators.valid_plugin_info_class(None))
+        self.assertTrue(PanoptesPluginInfoValidators.valid_plugin_info_class(PanoptesPollingPluginInfo))
+        self.assertTrue(PanoptesPluginInfoValidators.valid_plugin_info_class(PanoptesPluginInfo))
+        self.assertFalse(PanoptesPluginInfoValidators.valid_plugin_info_class(PluginInfo))
+
+
+def mock_metadata_kv_store():
+    return MagicMock(side_effect=Exception)
+
+
+class TestPanoptesPluginInfo(unittest.TestCase):
+    @patch('redis.StrictRedis', panoptes_mock_redis_strict_client)
+    @patch('kazoo.client.KazooClient', panoptes_mock_kazoo_client)
+    def setUp(self):
+        self.my_dir, self.panoptes_test_conf_file = _get_test_conf_file()
+        self.__panoptes_context = PanoptesContext(self.panoptes_test_conf_file,
```

Comment:
Dunder names for properties is a bad pattern I started - let's not propagate it. Let's use a single underscore instead.